augurpy/calculate_AUC.py

```diff
@@ -1,14 +1,106 @@
-# This will be one of the main functions called by the user
-# It takes all the arguments like in the R version. 
-#
-# - check number of folds/subsample size are compatible (n > 1 in every fold)
-# - check augur mode -> run different function depending on mode
+from typing import Dict, List, Optional, Tuple, Union
 
+from anndata import AnnData
+from pandas import DataFrame
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.linear_model import LogisticRegression
 
-# data = read_load()
-# cv = cv()
-# scorer = scorer()
-# estimator = estimator()
 
-# calculate_AUC for each subsample using cross validation
-# (hier bin ich mir nicht sicher wo das hinmuss, muss auch nochmal nachlesen wie viel sklearn schon macht..)
\ No newline at end of file
+def calculate_auc(
+    input: Union[AnnData, DataFrame],
+    meta: Optional[DataFrame] = None,
+    label_col: str = "label",
+    cell_type_col: str = "cell_type",
+    n_subsamples: int = 50,
+    subsample_size: int = 20,
+    folds: int = 3,
+    min_cells: int = None,
+    var_quantile: float = 0.5,
+    feature_perc: float = 0.5,
+    n_threads: int = 4,
+    show_progress: bool = True,
+    augur_mode: str = "default",
+    classifier: str = "rf",
+    rf_params: Dict = None,
+    lr_params: List = None,
+) -> Tuple[Union[Dict, AnnData]]:
+    """Calculates the Area under the Curve using the given classifier.
+
+    Args:
+        input: Anndata or matrix containing gene expression values (genes in rows, cells in columns) and optionally meta
+            data about each cell.
+        meta: Optional data frame containing meta data about each cell.
+        label_col: column of the meta data frame or the anndata or matrix containing the condition labels for each cell
+            in the gene-by-cell expression matrix
+        cell_type_col: column of the meta data frame or the anndata or matrix containing the cell type labels for each
+            cell in the gene-by-cell expression matrix
+        n_subsamples: number of random subsamples to draw from complete dataset for each cell type
+        subsample_size: number of cells to subsample randomly per type from each experimental condition
+        folds: number of folds of cross-validati on to run
+        min_cells: minimum number of cells for a particular cell type in each condition in order to retain that type for
+            analysis (depricated..)
+        var_quantile: quantile of highly variable genes to retain for each cell type using the variable gene filter
+        feature_perc: proportion of genes that are randomly selected as features for input to the classifier in each
+            subsample using the random gene filter
+        n_threads: number of threads to use for parallelization
+        show_progress: if TRUE display a progress bar for the analysis with estimated time remaining
```

Comment:
```suggestion
        show_progress: if `True` display a progress bar for the analysis with estimated time remaining
```

Think this is nicer.